src/tactic/lint.lean

```diff
@@ -685,6 +685,23 @@ meta def should_be_linted (linter : name) (decl : name) : tactic bool := do
 e ← get_env,
 pure $ ¬ e.contains (mk_nolint_decl_name decl linter)
 
+/--
+`env.is_auto_decl d` returns true iff `d` is an automatically-generated
+declaration that should be ignored by (some) linters.
+-/
+meta def environment.is_auto_decl (env : environment) (d : declaration) : bool :=
```

Comment:
Move to `meta/expr.lean`? We should probably factor out `meta/environment.lean` at some point.